src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/handlers/CancellationCaseEventHandler.java

```diff
@@ -0,0 +1,201 @@
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.util.Arrays.asList;
+import static uk.gov.hmcts.reform.wacaseeventhandler.domain.camunda.DmnAndMessageNames.TASK_CANCELLATION;
+import static uk.gov.hmcts.reform.wacaseeventhandler.domain.camunda.DmnValue.dmnBooleanValue;
+import static uk.gov.hmcts.reform.wacaseeventhandler.domain.camunda.DmnValue.dmnStringValue;
+
+@Service
+@Order(1)
+@SuppressWarnings({"PMD.DataflowAnomalyAnalysis", "PMD.UseConcurrentHashMap", "unchecked"})
+public class CancellationCaseEventHandler implements CaseEventHandler {
+
+    private final AuthTokenGenerator serviceAuthGenerator;
+    private final WorkflowApiClient workflowApiClient;
+
+    public CancellationCaseEventHandler(AuthTokenGenerator serviceAuthGenerator, WorkflowApiClient workflowApiClient) {
+        this.serviceAuthGenerator = serviceAuthGenerator;
+        this.workflowApiClient = workflowApiClient;
+    }
+
+    @Override
+    public List<? extends EvaluateResponse> evaluateDmn(EventInformation eventInformation) {
+        String tableKey = TASK_CANCELLATION.getTableKey(
+            eventInformation.getJurisdictionId(),
+            eventInformation.getCaseTypeId()
+        );
+
+        String tenantId = eventInformation.getJurisdictionId().toLowerCase(Locale.ENGLISH);
+
+        EvaluateDmnRequest evaluateDmnRequest = buildEvaluateDmnRequest(
+            eventInformation.getPreviousStateId(),
+            eventInformation.getEventId(),
+            eventInformation.getNewStateId()
+        );
+
+        EvaluateDmnResponse<CancellationEvaluateResponse> response = workflowApiClient.evaluateCancellationDmn(
+            serviceAuthGenerator.generate(),
+            tableKey,
+            tenantId,
+            evaluateDmnRequest);
+
+        return response.getResults();
+    }
+
+    @Override
+    public void handle(List<? extends EvaluateResponse> results, EventInformation eventInformation) {
+        results.stream()
+            .filter(CancellationEvaluateResponse.class::isInstance)
+            .map(CancellationEvaluateResponse.class::cast)
+            .filter(result -> CancellationActions.CANCEL == CancellationActions.from(result.getAction().getValue()))
+            .forEach(cancellationEvaluateResponse -> {
+                DmnValue<String> taskCategories = cancellationEvaluateResponse.getTaskCategories();
+                DmnValue<String> processCategories = cancellationEvaluateResponse.getProcessCategories();
+                sendCancellationMessage(
+                    eventInformation.getCaseId(),
+                    taskCategories,
+                    processCategories
+                );
+            });
+    }
+
+    private EvaluateDmnRequest buildEvaluateDmnRequest(
+        String previousStateId,
+        String eventId,
+        String newStateId
+    ) {
+        Map<String, DmnValue<?>> variables = Map.of(
```

Comment:
Small suggestion: I think is more thread-safe and more performant to use the concurrent Hash map when possible. And you can get rid of that suppression at the top-level class too. 
I like to write  Map.of , though.